src/services/swap_service.ts

```diff
@@ -90,6 +93,53 @@ export class SwapService {
     private _altRfqMarketsCache: any;
     private _swapQuoter: SwapQuoter;
 
+    /**
+     * Returns an estimated price impact percent. This is estimated
+     * as the information used for calculation is based off of
+     * median values (fee sources) and not an exhaustive set of liquidity sources
+     * @param price the final price from the swap quote (inverted if buys)
+     * @param sellTokenToEthRate the rate of selling the sellToken to the native asset (e.g USDC->FTM)
+     * @param buyTokenToEthRate  the rate of selling the buy token to the native asset (e.g DAI->FTM)
+     * @param marketSide whether this is a sell or a buy (as the price is flipped)
+     * @returns an estimated price impact percentage calculated from the fee sources (median value)
+     */
+    private static _calculateEstimatedPriceImpactPercent(
+        price: BigNumber,
+        sellTokenToEthRate: BigNumber,
+        buyTokenToEthRate: BigNumber,
+        marketSide: MarketOperation,
+    ): BigNumber {
+        // ETH to USDC
+        // price: "2418.92"
+        // sellTokenToEthRate: "1"
+        // buyTokenToEthRate: "2438.74"
+
+        // If sell than price is in taker token, if buy price is inverted
+        const normalizedPrice = marketSide === MarketOperation.Sell ? price : new BigNumber(1).dividedBy(price);
+        // 2418.92 / (2438.74/1) = 0.99187 or 99.187%
+        const actualPriceToEstimatedPriceRatio = normalizedPrice.dividedBy(
+            buyTokenToEthRate.dividedBy(sellTokenToEthRate),
+        );
+        // 0.99187 -> 0.812%
+        const estimatedPriceImpactPercentage = new BigNumber(1)
+            .minus(actualPriceToEstimatedPriceRatio)
+            // tslint:disable:custom-no-magic-numbers
+            .times(100)
```

Comment:
NIT: While I personally prefer this percentage representation as it's hard to get it wrong. With this, we would have params like `slippagePercentage`, `buyTokenPercentageFee` that are 0 - 1, and this percentage response will be 0 - 100. Maybe we want to be consistent or name this slightly differently?